inference-engine/src/mkldnn_plugin/nodes/mkldnn_topk_node.h

```diff
@@ -4,111 +4,139 @@
 
 #pragma once
 
-#include "ie_common.h"
 #include <ie_common.h>
 #include <mkldnn_node.h>
+#include <string>
+#include <memory>
+#include <vector>
 
 namespace MKLDNNPlugin {
 
-class MKLDNNTopKNode : public MKLDNNNode {
-public:
-    MKLDNNTopKNode(const std::shared_ptr<ngraph::Node> &op, const mkldnn::engine &eng,
-                   MKLDNNWeightsSharing::Ptr &cache);
+enum TopKLayoutType {
+    topk_ncsp,
+    topk_nspc,
+    topk_blocked
+};
 
-    void getSupportedDescriptors() override {};
+enum TopKAlgorithm {
+    topk_bubble_sort,
+    topk_bitonic_sort,
+    topk_heap_sort
+};
 
-    void initSupportedPrimitiveDescriptors() override;
+struct jit_topk_config_params {
+    bool mode_max;
+    bool sort_index;
+    bool topk_innermost;
+    bool bubble_inplace; // all the elements in sorting is right in the register, no need to load and store for each comparison
+    TopKLayoutType layout;
+    TopKAlgorithm algorithm;
+    mkldnn::memory::data_type data_type;
+    InferenceEngine::Precision precision;
```

Comment:
We have data_type and precision which seems to mean the same thing. Can we keep only one?

Reply:
Fixed. Thanks Dmitry!